### ISSUE TITLE
datepicker - Auto-submit of the form when showing the calendar

### ISSUE THREAD

**Issue by extrafu** [angular/material#4192]
Hello,

Whenever a md-datepicker is added within a ```<form> ... </form>```, it auto-submits the form as soon as the arrow drop down icon is clicked, to actually show the calendar to select a day.

This happens with the latest code.

Thanks,

**Comment by epelc** [comment]
Perhaps `type="button"` needs to be added to the template.

**Comment by extrafu** [comment]
That patch fixes it for me:

```
diff --git a/src/components/datepicker/datePicker.js b/src/components/datepicker/datePicker.js
index 9dd7cd6..ec3e180 100644
--- a/src/components/datepicker/datePicker.js
+++ b/src/components/datepicker/datePicker.js
@@ -49,7 +49,7 @@
               'ng-class="{\'md-datepicker-focused\': ctrl.isFocused}">' +
             '<input class="md-datepicker-input" aria-haspopup="true" ' +
                 'ng-focus="ctrl.setFocused(true)" ng-blur="ctrl.setFocused(false)">' +
-            '<md-button md-no-ink class="md-datepicker-triangle-button md-icon-button" ' +
+            '<md-button md-no-ink class="md-datepicker-triangle-button md-icon-button" type="button" ' +
                 'ng-click="ctrl.openCalendarPane($event)" ' +
                 'aria-label="{{::ctrl.dateLocale.msgOpenCalendar}}">' +
               '<div class="md-datepicker-expand-triangle"></div>' +
```

**Comment by extrafu** [comment]
After further testing, sorry but it does not fix the issue.

**Comment by jelbourn** [comment]
Definitely should have `type="button"`

**Comment by jelbourn** [comment]
Fixed via 314a05df6be9d3136bb2594fd7357df033ee96fa
